### DISCUSSION TITLE
Generated js client - How to read enums and access the full enum list?

### DISCUSSION THREAD

**Discussion by danvuong** [connectrpc/connect-es#1028] [Q&A] [answered]
Hello everyone,

I have a JavaScript client generated with connect-es.
Some data properties returned by API are enums.

In 'car_pb.js':

```js
export const Car = proto3.makeMessageType(
  () => [
    { no: 1, name: "type",  kind: "enum", T: proto3.getEnumType(Car_Type) },
  ],
);
```

```js
export const Car_Type = proto3.makeEnum(
  [
    {no: 0, name: "TYPE_BLUE", localName: "BLUE"},
    {no: 1, name: "TYPE_RED", localName: "RED"},
  ],
);
```

When i get the data from the API I have only access to the enum Id or the lcal name (ex: "BLUE").
How can I access to the full name?

Also how can I get the full Car_Type definition from the generated client?

Thank you very much and sorry if this is not clear.

**Comment by srikrsna-buf** [comment]
Hey! the docs for the protobuf types are here: https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md#enumerations

**Reply by nicksnyder** [reply, marked as answer]
Updated link to documentation for enumerations: https://github.com/bufbuild/protobuf-es/blob/main/MANUAL.md#enumerations
